### PR TITLE
[DO NOT MERGE] Adding dependency example

### PR DESCRIPTION
**Pull request by dinoboff**
1. `jspm install d3`;
2. import d3 with `import * as d3 from 'd3';`;
3. removing d3 from from built app (in favour of a  cdn script) by editing [tools/bin/config.js](../tree/master/tools/bin/config.js) and [tools/assets/index.html](../tree/master/tools/assets/index.html)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
       "lib": "src/classmentors"
     },
     "dependencies": {
+      "d3": "npm:d3@^4.2.2",
```
Comment (dinoboff) [inline package.json:43]: Generated by `jspm install d3`

File: src/classmentors/index.js [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 import angular from 'angular';
 import module from 'classmentors/module.js';
+import * as d3 from 'd3';
+
+console.log(d3);
```
Comment (dinoboff) [inline src/classmentors/index.js:7]: To check it's working.

File: tools/bin/config.js [hunk 3]
```diff
@@ -42,6 +42,10 @@ const externals = [{
   name: 'angularfire',
   entry: 'angularfire',
   globalName: 'angular'
+}, {
+  name: 'd3',
+  entry: 'd3',
+  globalName: 'd3'
 }];
```
Comment (dinoboff) [inline tools/bin/config.js:49]: This array is use to build the options for `jspm build`, do define the libraries to exclude to define how the find them on the global scope.

See https://github.com/singpath/example-app/blob/master/docs/jspm.md#production-build

File: tools/assets/index.html [hunk 4]
```diff
@@ -11,6 +11,7 @@
     <script src="//cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.7/angular-messages.min.js"></script>
     <script src="//cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.7/angular-aria.min.js"></script>
     <script src="//cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.7/angular-animate.min.js"></script>
+    <script src="//cdnjs.cloudflare.com/ajax/libs/d3/4.2.2/d3.min.js"></script>
```
Comment (dinoboff) [inline tools/assets/index.html:14]: Script to load when d3 is excluding from the built app. (during development jspm/systemjs loads the one from "src/jspm_packages".

File: src/jspm.config.js [hunk 5]
```diff
@@ -138,11 +145,350 @@ SystemJS.config({
...
+    },
+    "github:jspm/nodelibs-stream@0.2.0-alpha": {
+      "map": {
+        "stream-browserify": "npm:stream-browserify@2.0.1"
+      }
+    },
+    "npm:stream-browserify@2.0.1": {
+      "map": {
+        "inherits": "npm:inherits@2.0.1",
+        "readable-stream": "npm:readable-stream@2.1.5"
```
Comment (dinoboff) [inline src/jspm.config.js:491]: Generated by `jspm install d3`.

**Comment by dinoboff** [comment]
@SingaporeClouds @shiquan-lim 

Anything unclear?

**Comment by shiquan-lim** [comment]
No, Damien. It looks ok for now. We will test using D3 in the application as well as perhaps adding new libraries and get back to you. Thank you.

**Comment by shiquan-lim** [comment]
@dinoboff I attempted to install angular-timer using bower, so that I can then subsequently export the module similar to how animate and material are exported. I received a permission denied error when running `bower install angular-timer` Please adivse how I may proceed

**Comment by dinoboff** [comment]
you need to install it with jspm like that:

```
jspm install angular-timer=github:siddii/angular-timer
```

I will check if it needs any package override .

**Comment by dinoboff** [comment]
In src/classmentors/index.js, import it and add "timer" to the angular module list to dependency

``` js
import 'angular-timer';

const module = angular.module('clm', [
  'angular-loading-bar',
  'firebase',
  'ngAnimate',
  'ngMessages',
  'ngRoute',
  'timer',
  spfShared.name
]);
```

**Comment by dinoboff** [comment]
ps: `jspm install angular-timer` doesn't work because the jspm registry doesn't define an alias for "github:sidii/angular-timer". 

So you need need to use `jspm install github:sidii/angular-timer`.

To simplify the import path, we create an alias with `jspm install angular-timer=github:sidii/angular-timer`.

Thankfully, the jspm registry already define the package override, and importing angular-timer will import its dependencies:

https://github.com/jspm/registry/blob/master/package-overrides/github/siddii/angular-timer%401.3.3.json

**Comment by dinoboff** [comment]
@shiquan-lim let me know if they any issue or anything unclear.

**Comment by shiquan-lim** [comment]
Wow thank you @dinoboff that was very comprehensive. I had tried using the jspm method but it had failed due to the alias issue you mentioned. I will attempt it and get back to you

**Comment by shiquan-lim** [comment]
@dinoboff I am still getting errors that moment and humanize-duration (angular-timer's dependencies), are not defined. I am trying to resolve this, but if you know the reason behind it that would be tremendously helpful. Thank you :)

**Comment by dinoboff** [comment]
I only checked it was loading and loading its dependencies. I will try to use it. The issue is probably with one of the dependencies override.

**Comment by dinoboff** [comment]
I opened an issue on your fork.

shiquan-lim/classmentors#236

**Pull request closed without merging**